tests/utils/test_matrix_operations.py

```diff
@@ -144,60 +95,254 @@ def test_csc_slice():
-    A1 = matrix_operations.slice_mat(A, np.array([0, 1]))
-    A2 = matrix_operations.slice_mat(A, 2)
-    A3 = matrix_operations.slice_mat(A, np.array([], dtype=int))
-    A4 = matrix_operations.slice_mat(A, np.array([0, 2], dtype=int))
-    A5 = matrix_operations.slice_mat(A, np.array([1, 1], dtype=int))
+# ------------------ Test matrix_slicer -----------------------
 
-    assert np.sum(A0 != A0_t) == 0
-    assert np.sum(A1 != A1_t) == 0
-    assert np.sum(A2 != A2_t) == 0
-    assert np.sum(A3 != A3_t) == 0
-    assert np.sum(A4 != A4_t) == 0
-    assert np.sum(A5 != A5_t) == 0
 
+def _get_matrix_slicer_target(mat, mode: Literal["float", "dense", "sparse", "ad"]):
+    """Get the target matrix for the matrix slicer test.
 
-def test_sliced_mat_rows():
-    A = sps.csr_matrix(np.array([[0, 0, 0], [1, 0, 0], [0, 0, 3]]))
+    Parameters:
+        mat: The matrix to be sliced
 
-    A0_t = sps.csr_matrix(np.array([[1, 0, 0], [0, 0, 3]]))
-    A1_t = sps.csr_matrix(np.array([[0, 0, 0], [1, 0, 0]]))
-    A2_t = sps.csr_matrix(np.array([[0, 0, 3]]))
-    A3_t = sps.csr_matrix(np.atleast_2d(np.array([[], [], []])).T)
-    A4_t = sps.csr_matrix(np.array([[0, 0, 0], [0, 0, 3]]))
-    A5_t = sps.csr_matrix(np.array([[1, 0, 0], [1, 0, 0]]))
-
-    A0 = matrix_operations.slice_mat(A, np.array([1, 2], dtype=int))
-    A1 = matrix_operations.slice_mat(A, np.array([0, 1]))
-    A2 = matrix_operations.slice_mat(A, 2)
-    A3 = matrix_operations.slice_mat(A, np.array([], dtype=int))
-    A4 = matrix_operations.slice_mat(A, np.array([0, 2], dtype=int))
-    A5 = matrix_operations.slice_mat(A, np.array([1, 1], dtype=int))
-
-    assert np.sum(A0 != A0_t) == 0
-    assert np.sum(A1 != A1_t) == 0
-    assert np.sum(A2 != A2_t) == 0
-    assert np.sum(A3 != A3_t) == 0
-    assert np.sum(A4 != A4_t) == 0
-    assert np.sum(A5 != A5_t) == 0
+    Returns:
+        The target quantity to be sliced.
+
+    """
+    if mode == "sparse":
+        target = mat
+    elif mode == "float":
+        return 42.0
+    else:
+        vec = np.array([1, 2, 3, 4, 5])
+        if mode == "dense":
+            # The dense mode uses a vector. In principle, it is possible also to
+            # consider 2d numpy arrays, but the MatrixSlicer is not designed to handle
+            # this case.
+            target = vec
+        else:
+            target = pp.ad.AdArray(vec, mat)
+    return target
+
+
+@pytest.mark.parametrize("mode", ["dense", "sparse", "ad"])
+@pytest.mark.parametrize(
+    "domain_inds, range_inds",
+    [
+        (np.array([3, 1]), np.array([0, 3])),
+        (np.array([3, 1]), np.array([0, 4])),
+        (np.array([2, 3, 0]), np.array([3, 0, 1])),
+        (None, np.array([0, 1, 3])),
+        (np.array([0, 1, 3]), None),
+    ],
+)
+@pytest.mark.parametrize("range_size", [None, 6])
+@pytest.mark.parametrize("domain_size", [None, 6])
+@pytest.mark.parametrize("transpose", [True, False])
+def test_matrix_slicer(
```

Comment:
This test could do with some more documentation.

Reply:
Updated documentation. Please let me know if you want more information.